src/trace/entry.zig

```diff
@@ -1,245 +1,83 @@
-            .noop => {},
-            .channel_print => |impl| impl.deinit(),
-            .direct_print => |impl| impl.deinit(),
+        pub inline fn init(logger: ScopedLogger(scope), level: Level) Self {
+            return .{ .logger = logger, .level = level, .fields = .{} };
         }
-    }
 
-    pub fn field(self: Self, name: []const u8, value: anytype) Self {
-        switch (self) {
-            .noop => {
-                return self;
-            },
-            .channel_print => |entry| {
-                var log_entry = entry;
-                _ = log_entry.field(name, value);
-                return Entry{ .channel_print = log_entry };
-            },
-            .direct_print => |entry| {
-                var log_entry = entry;
-                _ = log_entry.field(name, value);
-                return Entry{ .direct_print = log_entry };
-            },
+        /// Add a field to the log message.
+        pub inline fn field(
+            self: Self,
+            comptime name: [:0]const u8,
+            value: anytype,
+        ) Entry(FieldsPlus(name, @TypeOf(value)), scope) {
+            if (self.logger == .noop) return .{
+                .logger = .noop,
+                .level = undefined,
+                .fields = undefined,
+            };
+            var new_fields: FieldsPlus(name, @TypeOf(value)) = undefined;
+            inline for (@typeInfo(Fields).Struct.fields) |existing_field| {
+                @field(new_fields, existing_field.name) = @field(self.fields, existing_field.name);
+            }
+            @field(new_fields, name) = value;
+            return .{
+                .logger = self.logger,
+                .level = self.level,
+                .fields = new_fields,
+            };
         }
-    }
 
-    pub fn log(self: Self, comptime msg: []const u8) void {
-        switch (self) {
-            .noop => {},
-            .channel_print => |impl| {
-                var logger = impl;
-                logger.log(msg);
-            },
-            .direct_print => |impl| {
-                var logger = impl;
-                logger.log(msg);
-            },
+        /// Log the message using the logger, including all fields that are saved in the entry.
+        pub inline fn log(self: Self, comptime message: []const u8) void {
+            self.logger.private_log(self.level, self.fields, message, .{});
         }
-    }
 
-    pub fn logf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            .noop => {},
-            .channel_print => |impl| {
-                var logger = impl;
-                logger.logf(fmt, args);
-            },
-            .direct_print => |impl| {
-                var logger = impl;
-                logger.logf(fmt, args);
-            },
+        /// Log the message using the logger, including all fields that are saved in the entry.
+        pub inline fn logf(self: Self, comptime fmt: []const u8, args: anytype) void {
```

Comment:
Shouldn't be `inline`.

Reply:
f0c6b320cbe51686c31cf5e6ffffadd2325d52d7